Source/Fuse.Scripting.JavaScript/ThreadWorker.uno

```diff
@@ -140,21 +133,24 @@ namespace Fuse.Scripting.JavaScript
 		}
 
 		/*
-			Throws an exception that was generated on the thread. If there are more than one then the previous
-			ones will be reported and the last one thrown.
+			Dispatches an exception from the JS-thread to the UI-thread.
 		*/
-		public void CheckAndThrow()
+		void DispatchException(Exception e)
 		{
-			Exception next = null, prev = null;
-			while (_exceptionQueue.TryDequeue(out next))
-			{
-				if (prev != null)
-					Fuse.Diagnostics.UnknownException("Skipped Exception", prev, this);
-				prev = next;
-			}
+			_exceptionQueue.Enqueue(e);
```

Comment:
This could probably be simplified. Instead of using a queue we could make a closure on the exception and post an action on that directly. It's a more straight forward 1:1 relationship and avoids the queue.

Reply:
Yeah, I considered that, but it introduces new allocations. I don't really mind either way, though, as this only happens in error-cases. I can change if you think that'd be better... I don't really care.

Reply:
I think it might be better for a future reader of this code. The queue use is a bit odd now.

If you derive the closure from WrapException you could avoid any new allocation, but for this error path that's like the least of concerns I think.

Reply:
OK, fix incoming ;)

Reply:
@mortoray: Done :)